### PR TITLE
Migration Guide for version 2.4.0

### PR DESCRIPTION
**Pull request by a-szotyori**
Updated our Sphinx documentation and added a migration guide for version 2.4.0. Also added two util classes to the public API for aiding our clients in working with Intents and Uris with document import.

### How to test
Run `$ ginivision/src/doc/make html` to create the Sphinx documentation and `$ open ginivision/src/doc/build/html/index.html` to view it.
Run `$ ./gradlew ginivision:generateJavadoc` to create the Javadoc and `$ open ginivision/build/docs/javadoc/index.html` to view it.

### Merging
Automatic.

### CODE REVIEW

File: ginivision/src/doc/source/supporting-tablets.rst [hunk 1]
```diff
@@ -1,65 +0,0 @@
...
-The Gini Vision Library can be used on tablets, too. Some UI elements have been adapted to offer the best user experience for tablet users. Requirements and resources were also adapted.
-
-You may skip to the `Quick Checklist`_ to get an overview of the steps required for supporting tablets.
-
-Extraction Quality Considerations
----------------------------------
-
-We recommend implementing checks on tablet hardware to ensure that devices meet the Gini Vision Libraries minimum recommended hardware specifications.
-
-Many tablets with at least 8MP cameras don't have an LED flash (like the popular Samsung Galaxy Tab S2) therefore we don't require flash for tablets. For this reason the extraction quality on those tablets might be lower compared to smartphones.
```
Comment (mpoprat) [inline ginivision/src/doc/source/supporting-tablets.rst]: Many tablets with at least 8MP cameras don't have an LED flash (like the popular Samsung Galaxy Tab S2). Therefore, we don't require flash ...

File: ginivision/src/doc/source/supporting-tablets.rst [hunk 2]
```diff
@@ -1,65 +0,0 @@
...
-
-Hardware Requirements
----------------------
-
-We disabled the camera flash requirement for tablets. Camera flash is not a standard feature for tablets and even some popular models like the Samsung Galaxy Tab S2 don't have an LED flash.
-
-You can view the Gini Vision Library's hardware requirements `here <http://developer.gini.net/gini-vision-lib-android/javadoc/net/gini/android/vision/requirements/RequirementId.html>`_.
```
Comment (mpoprat) [inline ginivision/src/doc/source/supporting-tablets.rst]: Can you specify in the hardware requirements that flash on is only required for tablets?

File: ginivision/src/doc/source/supporting-tablets.rst [hunk 3]
```diff
@@ -1,65 +0,0 @@
...
-
-Supporting All Orientations
```
Comment (mpoprat) [inline ginivision/src/doc/source/supporting-tablets.rst]: Supporting All *Screen* Orientations

File: ginivision/src/doc/source/supporting-tablets.rst [hunk 4]
```diff
@@ -1,65 +0,0 @@
...
-
-* ``gv_camera_preview_corners.png`` - Document corner guides
-* ``gv_camera_preview_corners_land.png`` - Document corner guides for landscape (it's sufficient to use the portrait one rotated by 90 degrees)
-* ``gv_onboarding_flat.png`` - First onboarding page image
-* ``gv_onboarding_parallel.png`` - Second onboarding page image
-* ``gv_onboarding_align.png`` - Third onboarding page image
-
-These images are higher resolution versions of the same images that are used for phones.
-
-Quick Checklist
```
Comment (mpoprat) [inline ginivision/src/doc/source/supporting-tablets.rst]: - I would start the migration guide with the quick checklist so one can get an idea what to do. Then continue with the details.
- Also add the no-flash-requirement for tablets in the checklist

File: ginivision/src/doc/source/supporting-tablets.rst (hunk 2 above)
Comment (a-szotyori) [reply ginivision/src/doc/source/supporting-tablets.rst]: That link points to the live version which doesn't have tablets. The new javadoc mentions that:
screen shot 2017-10-24 at 16 40 33

File: ginivision/src/doc/source/supporting-tablets.rst (hunk 4 above)
Comment (a-szotyori) [reply ginivision/src/doc/source/supporting-tablets.rst]: This quick checklist ist for tablets only, I changed the title to Tablet Support Quick Checklist. I added the no flash requirement.

File: ginivision/src/doc/source/updating-to-2-4-0.rst [hunk 5]
```diff
@@ -0,0 +1,445 @@
...
+Additional methods were added to the ``Document`` to identify whether the document was imported, to get the ``Intent`` with which it was imported, to find out if it's reviewable (PDFs are not reviewable for ex.) and to get it's type (image or pdf).
+
+Use the additional methods in the Review and Analysis screens if you wish to handle imported documents separately from pictures taken by the camera.
+
+.. _Document Import - Analysis Screen for PDFs:
+
+Analysis Screen for PDFs
+~~~~
+
+The Analysis Screen will render the first page of the PDF on Android Lollipop and newer. On older versions no preview is shown. In addition above the PDF preview area the PDF filename is displayed and the nr of pages (on Android Lollipop and newer).
```
Comment (mpoprat) [inline ginivision/src/doc/source/updating-to-2-4-0.rst]: Can you add a version number to Lollipop, please? It makes it easier for me to communicate with the developers ...

**Pull request merged**